js/battle.js

```diff
@@ -5566,6 +5563,7 @@ var Battle = (function () {
 		}
 	};
 	Battle.prototype.run = function (str, preempt) {
+		var takesKwargs = {swap:1, move:1, cant:1, callback:1};
```

Comment:
I would rather have a whitelist of `nokwargs`.

Honestly, I'd rather go to a different approach altogether. We might want `|` as well as `[` as valid characters (e.g. in `|c|`), so it makes more sense to slice off the end of the string for a final argument of certain commands somehow.